wagtail/admin/views/page_privacy.py

```diff
@@ -27,10 +27,13 @@ def set_privacy(request, page_id):
             if form.cleaned_data['restriction_type'] == PageViewRestriction.NONE:
                 # remove any existing restriction
                 if restriction:
-                    restriction.delete()
+                    restriction.delete(user=request.user)
             else:
                 restriction = form.save(commit=False)
                 restriction.page = page
+                # pass the current user as restriction attribute since the model form calls a plain instance.save()
+                # the attribute is removed in PageViewRestriction.save()
+                restriction.user = request.user
```

Comment:
I'm pretty sure the only thing `commit=False` doesn't do is run `instance.save()` so could we just call `restriction.save()` here?